assets/js/admin/course-theme-sidebar.js

```diff
@@ -3,49 +3,152 @@
  */
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
 import { __ } from '@wordpress/i18n';
-import { SelectControl } from '@wordpress/components';
+import { ToggleControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import useCourseMeta from '../../react-hooks/use-course-meta';
+import { name as courseOutlineBlockName } from '../../blocks/course-outline/outline-block/block.json';
+import { name as courseModuleBlockName } from '../../blocks/course-outline/module-block/block.json';
+import { name as courseLessonBlockName } from '../../blocks/course-outline/lesson-block/block.json';
+import { getFirstBlockByName } from '../../blocks/course-outline/data';
 
 const SENSEI_THEME = 'sensei-theme';
 const WORDPRESS_THEME = 'wordpress-theme';
+const SENSEI_PREVIEW_QUERY = 'sensei_theme_preview';
+
+const canPreview = ( block ) =>
+	block.name === courseLessonBlockName && block.attributes.id;
+
+const usePreviewAndCustomizerLinks = () => {
+	const currentPost = useSelect( ( select ) =>
+		select( 'core/editor' ).getCurrentPost()
+	);
+
+	const globalLearningModeEnabled =
+		window.sensei?.senseiSettings?.sensei_learning_mode_all || false;
+
+	const firstLesson = useSelect( ( select ) => {
+		const { getBlocks, getBlockAttributes } = select( 'core/block-editor' );
+		const blocks = getBlocks();
+		const courseOutline = getFirstBlockByName(
+			courseOutlineBlockName,
+			blocks
+		);
+
+		if ( ! courseOutline ) {
+			return {};
+		}
+
+		for ( const moduleOrLesson of getBlocks( courseOutline.clientId ) ) {
+			if ( canPreview( moduleOrLesson ) ) {
+				return getBlockAttributes( moduleOrLesson.clientId );
+			}
+
+			if ( moduleOrLesson.name === courseModuleBlockName ) {
+				for ( const lesson of getBlocks( moduleOrLesson.clientId ) ) {
+					if ( canPreview( lesson ) ) {
+						return getBlockAttributes( lesson.clientId );
+					}
+				}
+			}
+		}
+
+		return {};
+	} );
+
+	let previewUrl = '';
+	if ( firstLesson?.id && currentPost?.id ) {
+		if (
+			globalLearningModeEnabled ||
+			currentPost.meta._course_theme === SENSEI_THEME
+		) {
+			previewUrl = `/?p=${ firstLesson.id }`;
+		} else {
+			previewUrl = `/?p=${ firstLesson.id }&learn=1&${ SENSEI_PREVIEW_QUERY }=${ currentPost.id }`;
+		}
+
+		if ( firstLesson.draft ) {
+			previewUrl = `${ previewUrl }&post_type=lesson`;
+		}
+	}
+
+	let customizerUrl = '';
+	if ( previewUrl ) {
+		customizerUrl = `/wp-admin/customize.php?url=${ encodeURIComponent(
```

Comment:
This URL loaded the customizer on the correct page for me, but it would be nice if it would open directly in the "Sensei LMS > Course Theme" menu. Have you checked if it's possible?

Reply:
I found a link with some [instructions](https://wordpress.stackexchange.com/questions/214473/link-to-specific-customizer-section).
I didn't test it, but I hope it works. =)

Reply:
Updated! 🎉 Thanks for the tip. I was actually thinking about doing it. Kinda de-prioritized originally and then forgot about it. Now it works great. 👍 972fe20caaffb980c4897219031e6cd74324d007